### DISCUSSION TITLE
I got error at fund vrf subscription in code

### DISCUSSION THREAD

**Discussion by Vincent71399** [Cyfrin/foundry-full-course-cu#2173] [Q&A] [answered]
1. I followed the video and try to fund my created subscription with its Id
the code segment is as follows:

            vm.startBroadcast();
            LinkToken(linkToken).transferAndCall(vrfCoordinator, FUND_AMOUNT, abi.encode(subscriptionId));
            vm.stopBroadcast();

but I got error:
          
![image](https://github.com/user-attachments/assets/78d79dc6-4091-4808-881d-bb32dda8447b)

2. then I try to call the sepolia LINK token contract  https://sepolia.etherscan.io/address/0x779877A7B0D9E8603169DdbD7836e478b4624789
transferAndCall manually, with abi.encode my subscription Id, 
the vrfCoordinator address: 0x9DdfaCa8183c41ad55329BdeeD9F6A8d53168B1B
the transactions will be completed, my metamask wallet link balance will be deduced, but the link balance in my subscription will not be increased

anyone know why?

**Comment by EngrPips** [comment, marked as answer]
This is the first time I will be seeing a `Mac Mismatch` revert in the Foundry stack trace. That is strange to me. Maybe run `foundryup` and then retry. Also, you mentioned that you called the Sepolia link contract manually, and the balance on your subscription doesn't get increased still. That's not supposed to be the case. Check that you are doing everything correctly.

**Reply by Vincent71399** [reply]
forget it, I got the key password wrong

**Reply by EngrPips** [reply]
Interesting, I am glad you catch it hahaha.
